outlines/__init__.py

```diff
@@ -39,17 +33,3 @@
     "from_transformers",
     "from_vllm",
 ]
-
-__all__ = [
```

Comment:
So there's no `__all__` anymore at the root? If that's intended I guess we should also get rid of the unused `model_list` variable above.

Reply:
the only exception to this is the top-level `from outlines.models import *`. this will always add everything in the `outlines/models` package `__all__` to the top-level `outlines` import.

i put that there to align with the docs that suggest this type of usage:

```py
import outlines

outlines.from_<provider model>(...)
```

generally not advised to have wildcard exports but i think its ok so long as contributors follow the "only put public things in `__all__`" mentality (which is intuitive) and understand that everything public in the models package gets exposed top-level.

the alternative is a breaking change to make it `outlines.models.from_<provider model>(...)` and then disallowing any wildcard imports in root going forward.